apps/ui-mobile/android/app/src/main/java/com/uimobile/modules/MetaWearModule.java

```diff
@@ -18,70 +18,90 @@
+				led.editPattern(Led.Color.BLUE, Led.PatternPreset.BLINK)
+					.repeatCount((byte) reapeatTimes)
+					.commit();
+				led.play();
+			}
+		} else {
+			Log.i("MainActivity", "Board is not connected");
+		}
+	}
 
-  private void retrieveBoard(String metaWearMacAddress, Promise promise) {
-    final BluetoothManager btManager = this.application.getBluetoothManager();
-    final BluetoothDevice remoteDevice =
-      btManager.getAdapter().getRemoteDevice(metaWearMacAddress);
+	private void retrieveBoard(String metaWearMacAddress, Promise promise) {
+		final BluetoothManager btManager = this.application.getBluetoothManager();
+		final BluetoothDevice remoteDevice =
+			btManager.getAdapter().getRemoteDevice(metaWearMacAddress);
 
-    board = this.application.getMetaWearBoardFromDevice(remoteDevice);
-    Log.i("MainActivity", board.toString());
+		board = this.application.getMetaWearBoardFromDevice(remoteDevice);
+		this.application.setBoard(board);
+		Log.i("MainActivity", board.toString());
 
-    board.connectAsync().continueWith(new Continuation<Void, Void>() {
-      @Override
-      public Void then(Task<Void> task) throws Exception {
-        if (task.isFaulted()) {
-          Log.i("MainActivity", "Failed to connect");
-          promise.reject("Failed to connect", task.getError());
-        } else {
-          Log.i("MainActivity", "Connected");
-          promise.resolve("Connected");
-          board.readDeviceInformationAsync()
-            .continueWith(new Continuation<DeviceInformation, Void>() {
-              @Override
-              public Void then(Task<DeviceInformation> task) throws Exception {
-                Log.i("MainActivity", "Device Information: " + task.getResult());
-                return null;
-              }
-            });
-        }
-        return null;
-      }
-    });
+		board.connectAsync().continueWith(new Continuation<Void, Void>() {
+			@Override
+			public Void then(Task<Void> task) throws Exception {
+				if (task.isFaulted()) {
+					Log.i("MainActivity", "Failed to connect");
+					promise.reject("Failed to connect", task.getError());
+				} else {
+					Log.i("MainActivity", "Connected");
+					promise.resolve("Connected");
+					board.readDeviceInformationAsync()
+						.continueWith(new Continuation<DeviceInformation, Void>() {
+							@Override
+							public Void then(Task<DeviceInformation> task) throws Exception {
+								Log.i("MainActivity", "Device Information: " + task.getResult());
+								return null;
+							}
+						});
+				}
+				return null;
+			}
+		});
+	}
 
-  }
+	@ReactMethod
+	public void startMetaWearModules() {
+		accelerometerModule.startModule();
+		magnetometerModule.startModule();
+		gyroscopeModule.startModule();
+	}
 
+	@ReactMethod
+	public void stopMetaWearModules() {
+		accelerometerModule.stopModule();
+		magnetometerModule.stopModule();
+		gyroscopeModule.stopModule();
+		//application.getBoard().tearDown();
+	}
 }
```

Comment:
A w sumie takie pytanie z ciekawości, czy zrobienie tego reflesją tudzież podłączenie do bazowej klasy i odpalenie tego w pętli, byłoby rzucaniem sobie kłody pod nogi xD ? bo zastanawiam sie czy w wolnej chwili samemu by tego nie zrobic

Reply:
1. tutaj ten tearDown() do wywalenia/moze zostac zakomentowany.
2. nie czaje, po co w petli? ze iteracja po modulach?

Reply:
myślałem żeby zrobić to dynamiczne, tzn jak powstaną inne moduły do poboru danych z opaski to zeby tylko dodać do listy i te liste przekazać MetaWearModule o ona je już sobie obsłuzy m.in. wie ze kazda instancja w liscie bedzie miała metodę stopModule() i startModule().

Ale też pytanie czy my juz wszystkich modułów do pobierania danych juz nie mamy zaimplementowanych xD ?

Reply:
Można tak zrobic, mam dorobic w tym sprincie pobieranie danych z modulow 'forced' (cisnienie, temperatura itd.), wiec w sumie taka petla bedzie spoko. W praniu pewnie wyjda jakies problemy bo chyba jedna opsaka nie ma czesci modulow, wiec bedzie troche dlubania, zeby to obsluzyc odpowiednio, ale to juz nie bedzie kwestia petli tylko dostepow do modulow w opaskach. na razie nie ma co gdybac xD